File: frontend/backend/app/schemas/product.py
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel
from app.schemas.category import CategoryResponse

class ProductVariantBase(BaseModel):
    name: str
    price: float
    stock: int = 100
    sku: Optional[str] = None
    glitterOption: Optional[str] = None
    status: str = "ACTIVE"

class ProductVariantCreate(ProductVariantBase):
    pass

class ProductVariantResponse(ProductVariantBase):
    id: str
    productId: str
    createdAt: datetime
    updatedAt: datetime

    class Config:
        from_attributes = True

class ProductBase(BaseModel):
    name: str
    slug: str
    description: str
    shortDescription: Optional[str] = None
    categoryId: str
    price: float
    salePrice: Optional[float] = None
    sku: Optional[str] = None
    image: str
    imagesJson: Optional[str] = None
    status: str = "ACTIVE"
    isFeatured: bool = False
    inStock: bool = True
    isActive: bool = True
    advanceNoticeDays: int = 7
    advanceNoticeText: Optional[str] = "Make sure to place the order at least one week earlier."
    colorOptionAvailable: bool = True
    customizationAvailable: bool = True

class ProductCreate(ProductBase):
    variants: Optional[List[ProductVariantCreate]] = None

class ProductResponse(ProductBase):
    id: str
    category: Optional[CategoryResponse] = None
    variants: List[ProductVariantResponse] = []
    createdAt: datetime
    updatedAt: datetime

    class Config:
        from_attributes = True

File: frontend/types/product.ts
export interface Category {
  id: string;
  name: string;
  slug: string;
  description?: string;
  image?: string;
}

export interface ProductVariant {
  id: string;
  name: string;
  price: number;
  stock: number;
  sku: string;
}

export interface Product {
  id: string;
  name: string;
  slug: string;
  description: string;
  shortDescription?: string;
  categoryId: string;
  category?: Category;
  price: number;
  salePrice?: number;
  sku: string;
  image: string;
  images?: string[];
  isFeatured?: boolean;
  inStock?: boolean;
  isActive?: boolean;
  customizationAvailable?: boolean;
  colorOptionAvailable?: boolean;
  advanceNoticeDays?: number;
  variants?: ProductVariant[];
}
